agent/session.go

```diff
@@ -253,11 +253,24 @@ func (s *session) watch(ctx context.Context) error {
 		}
 		if tasksWatch != nil {
```

Comment:
Comment was added below indicating that it is indeed fallback code.  :)   `When falling back to Tasks because of an old managers, we wrap the tasks in assignments.`